File: Ahorn/triggers/CreditsTriggerPart2.jl
module CreditsTriggerPart2

using ..Ahorn, Maple

@mapdef Trigger "MaggyHelper/CreditsTriggerPart2" CreditsTriggerPart2(x::Integer, y::Integer, event::String="WaitJumpDash", height::Integer=16, width::Integer=16)

const placements = Ahorn.PlacementDict(
    "wait_jump_dash" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "wait_jump_double_dash" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "wait_jump_quintriple_dash" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "climb_down" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "wait" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "badeline_offset" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "chara_offset" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "kirby_offset" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "ralsei_offset" => Ahorn.EntityPlacement(CreditsTriggerPart2),
    "oshiro_marker" => Ahorn.EntityPlacement(CreditsTriggerPart2)
)

function Ahorn.selection(entity::CreditsTriggerPart2)
    return Ahorn.getEntityRectangle(entity)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::CreditsTriggerPart2, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.getEntityRectangle(entity), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/triggers/CreditsTriggerPart2.lua
local CreditsTriggerPart2 = {}

CreditsTriggerPart2.name = "MaggyHelper/CreditsTriggerPart2"
CreditsTriggerPart2.canResize = {true, true}
CreditsTriggerPart2.placements = {
    {
        name = "wait_jump_dash",
        data = {
            width = 16,
            height = 16,
            event = "WaitJumpDash"
        }
    },
    {
        name = "wait_jump_double_dash",
        data = {
            width = 16,
            height = 16,
            event = "WaitJumpDoubleDash"
        }
    },
    {
        name = "wait_jump_quintriple_dash",
        data = {
            width = 16,
            height = 16,
            event = "WaitJumpQuintripleDash"
        }
    },
    {
        name = "climb_down",
        data = {
            width = 16,
            height = 16,
            event = "ClimbDown"
        }
    },
    {
        name = "wait",
        data = {
            width = 16,
            height = 16,
            event = "Wait"
        }
    },
    {
        name = "badeline_offset",
        data = {
            width = 16,
            height = 16,
            event = "BadelineOffset"
        }
    },
    {
        name = "chara_offset",
        data = {
            width = 16,
            height = 16,
            event = "CharaOffset"
        }
    },
    {
        name = "kirby_offset",
        data = {
            width = 16,
            height = 16,
            event = "KirbyOffset"
        }
    },
    {
        name = "ralsei_offset",
        data = {
            width = 16,
            height = 16,
            event = "RalseiOffset"
        }
    },
    {
        name = "oshiro_marker",
        data = {
            width = 16,
            height = 16,
            event = "Oshiro"
        }
    }
}

CreditsTriggerPart2.fieldInformation = {
    event = {
        fieldType = "string",
        options = {
            "WaitJumpDash",
            "WaitJumpDoubleDash",
            "WaitJumpQuintripleDash",
            "ClimbDown",
            "Wait",
            "BadelineOffset",
            "CharaOffset",
            "KirbyOffset",
            "RalseiOffset",
            "Oshiro"
        },
        editable = false
    }
}

CreditsTriggerPart2.fieldOrder = {
    "x",
    "y",
    "width",
    "height",
    "event"
}

function CreditsTriggerPart2.selection(room, entity)
    return {
        entity.x,
        entity.y,
        entity.width or 16,
        entity.height or 16
    }
end

return CreditsTriggerPart2
